lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -305,3 +306,45 @@ def test_get_due_dates_error_fetching_block(self):
                 ]
                 actual_due_dates = get_due_dates(request, self.course.id, self.staff_user)
                 assert expected_due_dates == actual_due_dates
+
+
+class TestGetCourseMembers(CourseApiTestMixin, SharedModuleStoreTestCase):
+    """
+    Test get_course_members function
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestGetCourseMembers, cls).setUpClass()
+        cls.course = cls.create_course()
+        cls.honor = cls.create_user('honor', is_staff=False)
+        cls.staff = cls.create_user('staff', is_staff=True)
+        cls.instructor = cls.create_user('instructor', is_staff=True)
+
+        # Attach users with cls.course
+        cls.create_enrollment(user=cls.honor, course_id=cls.course.id)
+        cls.create_enrollment(user=cls.instructor, course_id=cls.course.id)
+        cls.create_courseaccessrole(user=cls.staff, course_id=cls.course.id, role='staff')
+        cls.create_courseaccessrole(user=cls.instructor, course_id=cls.course.id, role='instructor')
+
+    def test_get_course_members(self):
+        """
+        Test all different possible filtering
+        """
+        with self.assertNumQueries(3):
+            members = get_course_members(self.course.id)
+
+        self.assertEqual(len(members), 3)
+        self.assertEqual(
+            members[self.honor.id]['roles'],
+            ['student']
+        )
```

Comment:
We want to test that the combining of data works properly, so I think it would be useful to have test cases for people who exist only in enrollments (students), people who exist in both (course staff, beta testing students), and people who exist only in access_roles (data_researcher).